src/module-elasticsuite-catalog/Model/Healthcheck/CategoryIsAnchorConfigCheck.php

```diff
@@ -0,0 +1,290 @@
+
+    /**
+     * @var CategoryCollectionFactory
+     */
+    private $categoryCollectionFactory;
+
+    /**
+     * @var StoreManagerInterface
+     */
+    private $storeManager;
+
+    /**
+     * Constructor.
+     *
+     * @param Config                    $eavConfig                 EAV config.
+     * @param CategoryCollectionFactory $categoryCollectionFactory Category collection factory.
+     * @param StoreManagerInterface     $storeManager              Store manager.
+     * @param UrlInterface              $urlBuilder                URL builder.
+     * @param int                       $sortOrder                 Sort order (default: 70).
+     * @param int                       $severity                  Severity level.
+     */
+    public function __construct(
+        Config $eavConfig,
+        CategoryCollectionFactory $categoryCollectionFactory,
+        StoreManagerInterface $storeManager,
+        UrlInterface $urlBuilder,
+        int $sortOrder = 70,
+        int $severity = MessageInterface::SEVERITY_CRITICAL
+    ) {
+        parent::__construct($urlBuilder, $sortOrder, $severity);
+        $this->eavConfig = $eavConfig;
+        $this->categoryCollectionFactory = $categoryCollectionFactory;
+        $this->storeManager = $storeManager;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getIdentifier(): string
+    {
+        return 'category_is_anchor_config_check';
+    }
+
+    /**
+     * {@inheritDoc}
+     * @SuppressWarnings(PHPMD.ElseExpression)
+     * @throws LocalizedException
+     */
+    public function getDescription(): string
+    {
+        $defaultText = __(
+            'For showing products in categories, Elasticsuite relies on making sure all categories are "anchor" categories,
+             that is displaying both their "own" products and the products associated with their subcategories.'
+        );
+
+        $nonAnchorCount = $this->getNonAnchorCategoryCount();
+
+        // If the category configuration check failed.
+        if ($nonAnchorCount > 0) {
+            $nonAnchorCategories = $this->getNonAnchorCategoryCollection();
+
+            // If nonAnchorCategories ≤ 5 categories.
+            if ($nonAnchorCount <= self::MAX_DISPLAYED_NON_ANCHOR_CATEGORIES_LIMIT) {
+                $itemsList = $this->renderCategoriesList($nonAnchorCategories);
+
+                // @codingStandardsIgnoreStart
+                return implode(
+                    '<br />',
+                    [
+                        $defaultText,
+                        __(
+                            'Some of your categories are not configured as "anchor" categories, this may cause some unexpected issues in the catalog navigation and search result pages.'
+                        ),
+                        __(
+                            'This is the list of categories: %1',
+                            [$itemsList]
+                        ),
+                    ]
+                );
```

Comment:
Why returning here ?
Now you can't have both messages for categories with invalid attribute value and the attribute not having the correct default value ...

Reply:
Fixed.